velox/functions/lib/aggregates/DecimalAggregate.h

```diff
@@ -338,4 +338,37 @@ class DecimalAggregate : public exec::Aggregate {
   DecodedVector decodedPartial_;
 };
 
+/* This method is used to correct the sum result calculated using
+ * addWithOverflow. In the addWithOverflow method, when summing two int128_t
+ * with the same sign, it first converts them to uint128_t for calculation. The
+ * returned sum represents the lower 127 bits of the uint128_t sum. The overflow
+ * variable indicates whether an overflow occurred during the calculation. If
+ * overflow = 1, it means there was an upward overflow, and if overflow = -1,
+ * there was a downward overflow. However, since the sum only retains the lower
+ * 127 bits, it may miss one calculation of +(1 << 127) or -(1 << 127).
+ * Therefore, we need to make the following adjustments:
+ * 1. If overflow = 1 && sum < 0, it means that the calculation process missed
+ * one calculation of +(1 << 127), it needs to be corrected to
+ * +(1 << 127) + sum.
+ * 2. If overflow = -1 && sum > 0, it means that the calculation process missed
+ * one calculation of -(1 << 127), it needs to be corrected to
+ * -(1 << 127) + sum.
+ *
+ */
+inline static std::optional<int128_t> computeValidSum(
+    int128_t sum,
+    int64_t overflow) {
```

Comment:
Given that overflow has only 3 values: 0, 1, -1; perhaps, it would be more readable to use an enum.

Reply:
`overflow` can be greater than 1 or less than -1. In the actual calculation, the overflow in the accumulator accumulates continuously, but each time it can only increase by 1 or decrease by 1.

https://github.com/facebookincubator/velox/blob/fcb9a37b5c73068c5a43fd82e8b6cb7685a33cd7/velox/functions/lib/aggregates/DecimalAggregate.h#L327-L328

Reply:
OK. I will try to add some comments.